### PR TITLE
Remove lockservice

### PR DESCRIPTION
**Pull request by mystenmark**

### CODE REVIEW

File: crates/sui-core/src/authority/authority_store.rs [hunk 1]
```diff
@@ -861,58 +835,280 @@ impl AuthorityStore {
...
+    async fn acquire_transaction_locks(
+        &self,
+        epoch: EpochId,
+        owned_input_objects: &[ObjectRef],
+        tx_digest: TransactionDigest,
+    ) -> SuiResult {
+        // Other writers may be attempting to acquire locks on the same objects, so a mutex is
+        // required.
+        // TODO: replace with optimistic transactions (i.e. set lock to tx if none)
+        let _mutexes = self.acquire_locks(owned_input_objects).await;
```
Comment (mystenmark) [inline crates/sui-core/src/authority/authority_store.rs:887]: Note: this is one of the very few (maybe only) places in all this code that was modified when moving it over from lock_service.rs

File: crates/sui-core/src/authority/authority_store.rs [hunk 2]
```diff
@@ -861,58 +835,280 @@ impl AuthorityStore {
...
-        write_batch.write()?;
-        trace!("Finished writing batch");
-
-        // Need to have a critical section for now because we need to prevent execution of older
-        // certs which may overwrite newer objects with older ones.  This can be removed once we have
-        // an object storage supporting multiple object versions at once, then there is idempotency and
-        // old writes would be OK.
-        {
-            // Acquire the lock to ensure no one else writes when we are in here.
-            let _mutexes = self.acquire_locks(&owned_inputs[..]).await;
```
Comment (mystenmark) [inline crates/sui-core/src/authority/authority_store.rs:874]: note that I finally removed these mutexes, which we haven't needed for a long time.

**Review by sadhansood — APPROVED** [review]
🚀 Really excited to see that we are finally removing lockservice

**Review by lxfind — COMMENTED** [review]
LGTM. A few naming suggestions

**Pull request merged**